Dockerfile.base

```diff
@@ -0,0 +1,35 @@
+FROM ubuntu:16.04
+
+RUN apt-get update -y -qq
+RUN apt-get install -y -qq \
```

Comment:
I thought that libyaml-dev is must for `YAML`. ubuntu 16.10 uses ruby 2.0 or above? If so, libyaml-dev might be already installed.

> Easy enough to change the base container when needed.

Okay. We might need to run specs in container. I will research Travis CI and Circle CI if they can run specs in docker container, and will try to set up Wercker (wercker.com)